eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/SlugConverter.php

```diff
@@ -142,16 +180,23 @@ class SlugConverter
      */
     protected $transformationProcessor;
 
+    /**
+     * @var array
+     */
+    protected $configuration;
+
     /**
      * Creates a new URL slug converter.
      *
      * @param \eZ\Publish\Core\Persistence\TransformationProcessor $transformationProcessor
      * @param array $configuration
      */
-    public function __construct(TransformationProcessor $transformationProcessor, array $configuration = array())
-    {
+    public function __construct(
+        TransformationProcessor $transformationProcessor,
+        array $configuration = array()
+    ) {
         $this->transformationProcessor = $transformationProcessor;
-        $this->configuration = $configuration + $this->configuration;
+        $this->configuration = $configuration + self::DEFAULT_CONFIGURATION;
```

Comment:
This will only create a union on first level of the configuration, meaning we won't be able to add custom transformation groups, instead, the whole list will need to be redefined. I don't know if this is important or not.

Reply:
You mean, this were something different from current solution?
I was worried about BC breaks if someone injected that parameter, so right now you can add new (or  add new commands to existing ones) transformation groups via symantic configuration.

Reply:
Well, currently, if you implement a decorator for slug converter service, you can modify the configuration as you wish: adding commands to existing groups, adding new groups and so on.

With this PR, it wouldn't be possible any more, you would always need to redefine the entire groups config, even if you only want to add a single command to a group.

Reply:
@emodric unfortunately `SlugConverter` is internal (Core) service not meant to be extended. BC promise applies to API and SPI only.

That said, we're not sure if we understand you use case. Could you provide sample code that would break if we accepted this solution? 
Maybe we can find some other way to do it. 
However please keep in mind that configuring this service is handled by Semantic Configuration for which we have to provide BC.

Reply:
You misunderstood me. I'm not saying anything will break, because at worst, it will work as before. My point was that if you're implementing this feature, it might be nice to implement properly, to be able to add new transformation groups or add new commands to existing groups without having to override the entire configuration or having to write decorators for `SlugConverter`.

Reply:
And you can, via new symantic configuration.

Reply:
No problem, I'm just making sure it is possible to do it, nothing more, since we often have usecases where we need to add some commands to existing groups. `array_merge_recursive` in a compiler pass obviously does merge correctly, so it should work.

Reply:
I'm unsure how others feel about this, but if we want to let people add rules to existing groups then we might need to find a way to do that, deprecating old way of injecting config and exposing a new way that merges. /cc @bdunogier

Reply:
>but if we want to let people add rules to existing groups then we might need to find a way to do that, 

You can do that by semantic configuration right now. Right now, it just works both ways.

Reply:
Doc with current example configuration:
https://github.com/ezsystems/ezpublish-kernel/pull/2306/files#diff-78c39a773f9efe5f9c22eca3a9f246fbR578